### PR TITLE
improved error msg for invalid axis(`mx.split`)

### LINKED ISSUE #593: An enhancement of the error message for mlx.core.split
**Describe the bug**
When axis exceeds array dimensions, there should be a more explicit error message.

**To Reproduce**
```python
import mlx.core as mx

a_mx = mx.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=mx.float32)

indices_or_sections = 3
split = mx.split(a_mx, indices_or_sections, axis=2)
print([arr for arr in split])
```
image

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version 0.0.10

### PR DESCRIPTION
**Pull request by ManishAradwad**
## Proposed changes

closes #593.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: mlx/ops.cpp [hunk 1]
```diff
@@ -632,6 +632,13 @@ std::vector<array> split(
 
 std::vector<array>
 split(const array& a, int num_splits, int axis, StreamOrDevice s /* = {} */) {
+  auto ax = axis < 0 ? axis + a.ndim() : axis;
+  if (ax < 0 || ax >= a.ndim()) {
+    std::ostringstream msg;
+    msg << "Invalid axis(" << axis << ") passed to split"
```
Comment (awni) [inline mlx/ops.cpp:638]:
```suggestion
    msg << "Invalid axis " << axis << " passed to split"
```

File: python/tests/test_ops.py [hunk 2]
```diff
@@ -1027,6 +1027,9 @@ def test_split(self):
         self.assertEqual(y.tolist(), [[3, 4]])
         self.assertEqual(z.tolist(), [[5, 6]])
 
+        with self.assertRaises(ValueError):
+            _ = mx.split(a, 3, axis=2)
```
Comment (awni) [inline python/tests/test_ops.py:1031]:
```suggestion
           mx.split(a, 3, axis=2)
```

**Review by awni — COMMENTED** [review]
Looks great! I left a couple nits inline. Could you check then we can merge?

**Comment by ManishAradwad** [comment]
Done

**Comment by awni** [comment]
Looks like the formatting failed?

**Review by awni — APPROVED** [review]
Thanks!!

**Pull request merged**